advanced-solidus/images-and-image-processing.md

```diff
@@ -0,0 +1,196 @@
+
+{% code title="app/overrides/amazing_store/spree/image/custom_style.rb" %}
+```ruby
+# ...
+module AmazingStore
+  module Spree
+    module Image
+      module CustomStyle
+        def self.prepended(klass)
+          klass.attachment_definitions[:attachment][:styles] = {
+            mini: '48x48>',
+            small: '400x400>',
+            product: '680x680>',
+            large: '1200x1200>',
+            jumbo: '1600x1600>'
+          }
+        end
+
+        ::Spree::Image.prepend self
+      end
+    end
+  end
+end
+}
+```
+{% endcode %}
+
+Additionally, if further changes to the module are intended, we can simply overwrite the entire ActiveStorageAttachment/PaperclipAttachment modules by creating and alternate override:
+
+{% code title="app/models/amazing_store/spree/image/my_active_storage_attachment.rb" %}
+```ruby
+module AmazingStore::Spree::Image::MyActiveStorageAttachment
+  extend ActiveSupport::Concern
+  include ::Spree::ActiveStorageAdapter
+
+  delegate :width, :height, to: :attachment, prefix: true
+
+  included do
+    validates :attachment, presence: true
+    validate :attachment_is_an_image
+    validate :supported_content_type
+
+    has_attachment :attachment,
+                   styles: {
+                    mini: '48x48>',
+                    small: '400x400>',
+                    product: '680x680>',
+                    large: '1200x1200>',
+                    jumbo: '1600x1600>'
+                   },
+                   default_style: :product
+
+    def supported_content_type
+      unless attachment.content_type.in?(::Spree::Config.allowed_image_mime_types)
+        errors.add(:attachment, :content_type_not_supported)
+      end
+    end
+  end
+end
+```
+{% endtab %}
+
+However, to make the module override work, we will also have to instruct Spree to utilize our customized `my_active_storage_attachment` instead of the default:
+
+{% code title="config/initializers/spree.rb" %}
+```ruby
+Spree.config do |config|
+  # ...
+  config.image_attachment_module = 'AmazingStore::Spree::Image::ActiveStorageAttachment'
+  # ...
+end
+```
+{% endcode %}
+
+After appending these new styles into settings, they can be utilized in your store by passing the style when rendering the image partial. Ex:
+
+{% code title="" %}
+```ruby
+<%= render 'spree/admin/shared/image', image: product.gallery.images.first, size: :jumbo %>
+```
+{% endcode %}
+
+## Modifying Variant Options
+### ActiveStorage
+
+MiniMagick allows the processing of variants with many [options](https://github.com/janko/image_processing/blob/master/doc/minimagick.md#imageprocessingminimagick) to customize your images. To take full advantage of these options, we will need to inject our modified method for `variant`. This can be accomplished with an **override** to Spree's current `variant` method utilizing Rail's suggested pattern for [improving engine functionality](https://guides.rubyonrails.org/engines.html#improving-engine-functionality).
```

Comment:
I don't think we can break `Rail's` this way, I never seen it used like this, but of course, I could be wrong here. 😄

Reply:
Thanks for the catch! corrected the grammar error